internal/pkg/resourceconfig/resourceconfig.go

```diff
@@ -135,29 +134,51 @@ func (p *RawConfigFileProvider) IsSupported(path string) bool {
 }
 
 // GetConfig returns the resource configs
-func (p *RawConfigFileProvider) GetConfig(path string) ([]*unstructured.Unstructured, error) {
+// from a directory or a file containing raw Kubernetes resource configurations
+func (p *RawConfigFileProvider) GetConfig(root string) ([]*unstructured.Unstructured, error) {
 	var values clik8s.ResourceConfigs
 
-	b, err := ioutil.ReadFile(path)
-	if err != nil {
-		return nil, err
-	}
-	objs := strings.Split(string(b), "---")
-	for _, o := range objs {
-		body := map[string]interface{}{}
+	err := filepath.Walk(root, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
we should only load yaml files with a gvk.

Reply:
added functions `isYamlFile` and `hasApiVersionKind`. 
Also covered these cases in tests:
- a readme.md file
- a YAML file without `apiVersion` or `kind`